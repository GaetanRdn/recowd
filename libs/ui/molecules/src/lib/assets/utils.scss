@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
