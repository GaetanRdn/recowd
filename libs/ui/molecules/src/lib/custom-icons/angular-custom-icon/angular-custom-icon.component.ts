import { Component } from '@angular/core';
import { CustomIconComponent } from '@recowd/ui-atoms';

@Component({
  selector: 'rc-angular-custom-icon',
  standalone: true,
  imports: [CustomIconComponent],
  templateUrl: './angular-custom-icon.component.html',
})
export class AngularCustomIconComponent extends CustomIconComponent {
  public override readonly name = 'assets/images/angular';
}
