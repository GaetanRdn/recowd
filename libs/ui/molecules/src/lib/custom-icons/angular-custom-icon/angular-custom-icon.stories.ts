import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { AngularCustomIconComponent } from './angular-custom-icon.component';

export default {
  title: 'molecules/custom-icons/angular-custom-icon',
  parameters: {
    jest: 'angular-custom-icon',
  },
  decorators: [
    moduleMetadata({
      imports: [AngularCustomIconComponent],
    }),
  ],
} as Meta<AngularCustomIconComponent>;

const Template: Story<AngularCustomIconComponent> = () => ({
  template: `<rc-angular-custom-icon></rc-angular-custom-icon>`,
});

export const Default = Template.bind({});
