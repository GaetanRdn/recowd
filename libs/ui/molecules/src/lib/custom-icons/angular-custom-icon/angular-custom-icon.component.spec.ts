import { TestBed, waitForAsync } from '@angular/core/testing';

import { AngularCustomIconComponent } from './angular-custom-icon.component';
import { Component } from '@angular/core';
import { TemplateLookup } from '@recowd/test/utils';

describe('AngularCustomIconComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [AngularCustomIconComponent],
    }).compileComponents();

    templateLookup = new TemplateLookup(HostComponent);
  }));

  it.each`
    width | height
    ${32} | ${undefined}
    ${16} | ${24}
  `(
    'should create with name $name, width $width and height $height',
    ({ width, height }) => {
      templateLookup.hostComponent.width = width;
      templateLookup.hostComponent.height = height;

      templateLookup.detectChanges();

      expect(templateLookup.firstChildElement).toMatchSnapshot();
    }
  );
});

@Component({
  template: ` <rc-angular-custom-icon
    [width]="width"
    [height]="height"
  ></rc-angular-custom-icon>`,
})
class HostComponent {
  public width!: AngularCustomIconComponent['width'];
  public height!: AngularCustomIconComponent['height'];
}
