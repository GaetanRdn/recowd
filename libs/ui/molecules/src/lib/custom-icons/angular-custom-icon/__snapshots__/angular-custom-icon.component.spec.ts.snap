// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AngularCustomIconComponent should create with name $name, width 16 and height 24 1`] = `
<rc-angular-custom-icon>
  <rc-custom-icon>
    <img
      alt="angular"
      height="24"
      src="assets/images/angular.svg"
      width="24"
    />
  </rc-custom-icon>
</rc-angular-custom-icon>
`;

exports[`AngularCustomIconComponent should create with name $name, width 32 and height undefined 1`] = `
<rc-angular-custom-icon>
  <rc-custom-icon>
    <img
      alt="angular"
      height="24"
      src="assets/images/angular.svg"
      width="24"
    />
  </rc-custom-icon>
</rc-angular-custom-icon>
`;
